odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/repository/reactive/ReactiveDataEntityQueryExampleRelationRepositoryImpl.java

```diff
@@ -99,15 +95,26 @@ public Flux<DataEntityToQueryExamplePojo> removeRelationWithDataEntityByQueryId(
 
     @Override
     public Flux<QueryExampleDto> getQueryExampleDatasetRelationsByDataEntity(final Long dataEntityId) {
-        final SelectHavingStep<Record> query = DSL.select(QUERY_EXAMPLE.asterisk())
-            .select(jsonArrayAgg(field(DATA_ENTITY.asterisk().toString())).as(AGG_DATA_ENTITIES_FIELD))
-            .from(QUERY_EXAMPLE)
-            .join(DATA_ENTITY_TO_QUERY_EXAMPLE)
-            .on(DATA_ENTITY_TO_QUERY_EXAMPLE.QUERY_EXAMPLE_ID.eq(QUERY_EXAMPLE.ID))
-            .join(DATA_ENTITY)
-            .on(DATA_ENTITY.ID.eq(DATA_ENTITY_TO_QUERY_EXAMPLE.DATA_ENTITY_ID))
-            .where(DATA_ENTITY_TO_QUERY_EXAMPLE.DATA_ENTITY_ID.eq(dataEntityId))
-            .groupBy(QUERY_EXAMPLE.ID);
+        final SelectConditionStep<Record1<Long>> queryExampleSelect =
+            DSL.select(DATA_ENTITY_TO_QUERY_EXAMPLE.QUERY_EXAMPLE_ID)
+                .from(DATA_ENTITY_TO_QUERY_EXAMPLE)
+                .where(DATA_ENTITY_TO_QUERY_EXAMPLE.DATA_ENTITY_ID.eq(dataEntityId));
+
+        final Table<Record1<Long>> exampleCTE = queryExampleSelect.asTable(QUERY_EXAMPLES_CTE);
+
+        final SelectHavingStep<Record> query =
+            DSL.with(exampleCTE.getName())
+                .as(queryExampleSelect)
```

Comment:
I'm not sure but It feels like this intermidiate step is unecessary.
```suggestion
    Table<Record1<Long>> exampleCTE = DSL.select(DATA_ENTITY_TO_QUERY_EXAMPLE.QUERY_EXAMPLE_ID)
       .from(DATA_ENTITY_TO_QUERY_EXAMPLE)
       .where(DATA_ENTITY_TO_QUERY_EXAMPLE.DATA_ENTITY_ID.eq(dataEntityId))
       .asTable(QUERY_EXAMPLES_CTE);
       
    SelectHavingStep<Record> query = DSL.with(exampleCTE)
```
It seems that this construct
```
DSL.with(exampleCTE.getName())
                .as(queryExampleSelect)
```
should behave the same way as this one
```
DSL.with(exampleCTE)
```

Am I missing something?

Reply:
We cannot pass simple exampleCTE into with because there no with method with this arguments like Table<Record1<Long>> exampleCTE.